src/content-handlers/iiif/modules/uv-mediaelementcenterpanel-module/MediaElementCenterPanel.ts

```diff
@@ -59,12 +59,32 @@ export class MediaElementCenterPanel extends CenterPanel<
     });
 
     this.extensionHost.subscribe(IIIFEvents.SET_TARGET, (target: TFragment) => {
-      let t = target.t;
+      let t: number | [number, number] = target.t;
       if (Array.isArray(t)) {
-        t = t[0];
+        if ((t as [number] | [number, number]).length === 1) {
+          t = t[0];
+        } else {
+          const [startTime, endTime] = t;
+
+          if (endTime <= startTime) {
+            console.error("endTime must be greater than startTime");
+            return;
+          }
+
+          this.player.setCurrentTime(startTime);
+          this.player.play();
+
+          const duration = (endTime - startTime) * 1000;
+
+          setTimeout(() => {
+            this.player.pause();
+          }, duration);
```

Comment:
Maybe need to cancel this timeout if `SET_TARGET` is called again, or if it's manually paused.

Reply:
Good point. And what should the behavior be if you unpause after pausing? Do we still expect the range to be respected, or does interrupting it cause the full media to play through?

Reply:
Have added some new logic to clear the timeout on new SET_TARGET or pause. Not sure what to do if unpaused again... I can live with it playing through to the end.
I've also added an autoPlayOnSetTarget config option which is necessary for my use case.